isthmus/src/main/java/io/substrait/isthmus/SqlExpressionToSubstrait.java

```diff
@@ -0,0 +1,114 @@
+package io.substrait.isthmus;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.substrait.extendedexpression.ExtendedExpressionProtoConverter;
+import io.substrait.extendedexpression.ImmutableExpressionReference;
+import io.substrait.extendedexpression.ImmutableExtendedExpression;
+import io.substrait.isthmus.expression.RexExpressionConverter;
+import io.substrait.isthmus.expression.ScalarFunctionConverter;
+import io.substrait.proto.ExtendedExpression;
+import io.substrait.type.NamedStruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql2rel.SqlToRelConverter;
+import org.apache.calcite.sql2rel.StandardConvertletTable;
+
+public class SqlExpressionToSubstrait extends SqlConverterBase {
+
+  public SqlExpressionToSubstrait() {
+    this(null);
+  }
+
+  protected SqlExpressionToSubstrait(FeatureBoard features) {
+    super(features);
+  }
+
+  private final ScalarFunctionConverter functionConverter =
+      new ScalarFunctionConverter(EXTENSION_COLLECTION.scalarFunctions(), factory);
+
+  private final RexExpressionConverter rexExpressionConverter =
+      new RexExpressionConverter(functionConverter);
```

Comment:
Small API consideration for a little future-proofing. I don't recommend hardcoding the `EXTENSION_COLLECTION` like this because it makes it difficult for users to provide their own extensions and/or manipulate existing ones (which is a limitation of the current `SqlToSubstrait` code).

What I recommend is something like
```java
    protected final RexExpressionConverter rexConverter;

    public SqlExpressionToSubstrait() {
        this(FEATURES_DEFAULT, EXTENSION_COLLECTION);
    }

    public SqlExpressionToSubstrait(
            FeatureBoard features,
            SimpleExtension.ExtensionCollection extensions) {
        super(features);
        var scalarFunctionConverter = new ScalarFunctionConverter(extensions.scalarFunctions(), factory);
        this.rexConverter = new RexExpressionConverter(scalarFunctionConverter);
    }

```
which gives users a bit more control over extensions.

Reply:
Changed